File: Data Structures & Algorithms/decode-string/submission-2.java
class Solution {
    public static String decodeString(String s) {
        int num = 0;
        StringBuilder curr = new StringBuilder();
        Deque<Map.Entry<Integer, StringBuilder>> stack = new ArrayDeque<>();
        for (char c: s.toCharArray()) {
            if (Character.isDigit(c)) {
                num = num * 10 + (c - '0');
            } else if (c == '[') {
                stack.push(Map.entry(num, curr));
                curr = new StringBuilder();
                num = 0;
            } else if (c == ']') {
                var top = stack.pop();
                StringBuilder prev = top.getValue();
                int repeat = top.getKey();
                prev.repeat(curr, repeat);
                curr = prev;
            } else {
                curr.append(c);
            }
        }
        return curr.toString();
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for c in tokens:
            if c != '+' and c != '-' and c != '*' and c != '/':
                stack.append(int(c))
            else:
                num1 = stack.pop()
                num2 = stack.pop()

                if c == '+':
                    stack.append(num1 + num2)
                elif c == '-':
                    stack.append(num2 - num1)
                elif c == '*':
                    stack.append(num1 * num2)
                elif c =='/':
                    stack.append(int(float(num2) / num1))
        return stack.pop()